### PR TITLE
Use proper streams to handle output. Needed in Cucumber.js 7.2.1

### PR DESCRIPTION
**Pull request by aslakhellesoy**
Cucumber.js 7.2.1 now adds an event listener to the output stream (`stream.on`) which was causing a crash since none of the "streams" in `src/renderer/output/*.js` are proper streams.

I have made them so.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -9,8 +9,8 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, windows-latest]
-        node: [8.x, 10.x, 12.x, 14.x]
-        electron: [8.2.0, 9, 10, 11]
+        node: [12.x, 14.x, 16.x]
+        electron: [11, 12, 13]
```
Comment (jbpros) [inline .github/workflows/build.yml:13]: Can you explain why older versions of node and electron are not supported anymore?

File: features/support/world.js [hunk 2]
```diff
@@ -25,12 +25,12 @@ class CucumberElectronWorld {
...
     args[0] = path.resolve(__dirname + '/../../bin/' + args[0])
 
-    this.execResult = { stdout: '', stderr: '', output: '', exitCode: null, error: null }
+    this.execResult = { stdout: '', stderr: '', exitCode: null, error: null }
     this.printExecResult = () =>
       '------------------------------------\n' +
       `The process exited with code ${this.spawnedProcess.exitCode}\n` +
       '------------------------------------\n' +
-      `OUTPUT:\n${this.execResult.output}\n` +
+      `Exception:\n${(this.execResult.error && this.execResult.error.stack) || ''}\n` +
```
Comment (jbpros) [inline features/support/world.js:33]: Could we make this not appear at all if there is no exception, to avoid noise?

File: package.json [hunk 3]
```diff
@@ -25,23 +25,23 @@
...
   "dependencies": {
-    "ansi-to-html": "^0.6.14",
+    "ansi-to-html": "^0.6.15",
     "electron-window": "^0.8.1"
   },
   "peerDependencies": {
-    "@cucumber/cucumber": "^7.0.0",
-    "electron": ">=8.2.0 <12"
+    "@cucumber/cucumber": "^7.2.1",
+    "electron": "^13.0.1"
```
Comment (jbpros) [inline package.json:33]: This contradicts the README, doesn't it? Shouldn't it be 11+?

**Review by jbpros — CHANGES_REQUESTED** [review]
Thanks for the PR, Aslak.

**Comment by jbpros** [comment]
@romaingweb and I had a look at the failing test. Cucumber.js uses Commander to parse CLI options. We suspect [changes in version 7 of Commander](https://github.com/tj/commander.js/blob/master/CHANGELOG.md#changed-2) regarding error output to cause asynchronous issues on Windows. Commander is now using `stderr.write` to output errors instead of `console.error`. As documented in [Node.js's documentation](https://nodejs.org/dist/latest-v14.x/docs/api/process.html#process_a_note_on_process_i_o), output to TTYs on Windows is asynchronous, while it is synchronous on POSIX. Our hypothesis (untested) is that the process exits before the "unknown option" error gets a chance to be printed out on Windows.

Maybe it is a small issue in Commander, possibly fixed by using Commander's [configureOutput](https://github.com/tj/commander.js#override-exit-and-output-handling) in Cucumber.js.

In all cases, this is absolutely not cucumber-electron's responsibility and we are going to remove the test here.

**Comment by romaingweb** [comment]
We investigated a bit more this morning with @jbpros. On windows when using electron 11, the electron process seems to hang between scenarios.
A quick way to fix this would be to stop supporting electron 11. We're not sure if it's a good move though as 11 is still one of the most downloaded version on NPM.

**Comment by romaingweb** [comment]
I did a little more investigation this afternoon. I tried catching some errors in the streams to see if that is what's making cucumber hang like this without success.

Another hypothesis might be that an error message is occurring when launching cucumber-electron that we do not see but opens one of those nasty pop-ups with the error in it.
Next step is probably to try and run this on a windows machine to verify that hypothesis.

File: .github/workflows/build.yml (hunk 1 above)
Comment (aslakhellesoy) [reply .github/workflows/build.yml:13]: Several reasons:

* Older versions of Node required a hack in `assertStderrIncludes`
* Node 10 and older are [no longer supported](https://nodejs.org/en/about/releases/).
* There is no need to support older Electron versions

File: features/support/world.js (hunk 2 above)
Comment (aslakhellesoy) [reply features/support/world.js:33]: I'd prefer to keep it since it sometimes helps diagnosing weird errors.

File: package.json (hunk 3 above)
Comment (aslakhellesoy) [reply package.json]: Let's only support the latest version to avoid bugs in older versions.

**Pull request merged**